case_studies/dc2/DC2_bootstrap_generate_data.py

```diff
@@ -0,0 +1,177 @@
+# pylint: skip-file
+# flake8: noqa
+from os import environ
+from pathlib import Path
+
+import GCRCatalogs
+import pandas as pd
+import torch
+from hydra import compose, initialize
+from hydra.utils import instantiate
+
+from bliss.catalog import SourceType
+from bliss.surveys.dc2 import DC2, from_wcs_header_str_to_wcs
+
+
+def get_lsst_params(
+    lsst_catalog_tensors_dict, cur_image_wcs, image_lim, image_height_index, image_width_index
+):
+    lsst_ra = lsst_catalog_tensors_dict["ra"]
+    lsst_dec = lsst_catalog_tensors_dict["dec"]
+    lsst_pt, lsst_pr = cur_image_wcs.all_world2pix(lsst_ra, lsst_dec, 0)
+    lsst_pt = torch.from_numpy(lsst_pt)
+    lsst_pr = torch.from_numpy(lsst_pr)
+
+    lsst_plocs = torch.stack((lsst_pr, lsst_pt), dim=-1)
+    lsst_source_type = lsst_catalog_tensors_dict["truth_type"]
+    lsst_flux = lsst_catalog_tensors_dict["flux"]
+
+    x0_mask = (lsst_plocs[:, 0] > image_height_index * image_lim) & (
+        lsst_plocs[:, 0] < (image_height_index + 1) * image_lim
+    )
+    x1_mask = (lsst_plocs[:, 1] > image_width_index * image_lim) & (
+        lsst_plocs[:, 1] < (image_width_index + 1) * image_lim
+    )
+    lsst_x_mask = x0_mask * x1_mask
+    # filter r band
+    lsst_flux_mask = lsst_flux[:, 2] > 0
+    # filter supernova
+    lsst_source_mask = (lsst_source_type != 3).squeeze(1)
+    lsst_mask = lsst_x_mask * lsst_flux_mask * lsst_source_mask
+
+    lsst_plocs = lsst_plocs[lsst_mask, :] % image_lim
+    lsst_source_type = torch.where(
+        lsst_source_type[lsst_mask] == 2, SourceType.STAR, SourceType.GALAXY
+    )
+    lsst_flux = lsst_flux[lsst_mask, :]
+
+    return lsst_plocs, lsst_source_type, lsst_flux
+
+
+def generate_lsst_fullcat_files(
+    split_results_path, split_ids, lsst_catalog_tensors_dict, output_path
+):
+    output_path.mkdir(parents=True, exist_ok=True)
+    for split_id in split_ids:
+        with open(split_results_path / split_id, "rb") as split_result_file:
+            split_result = torch.load(split_result_file)
+        cur_image_wcs = from_wcs_header_str_to_wcs(split_result["wcs_header_str"])
+        image_lim = split_result["images"].shape[1]
+        assert (
+            split_result["images"].shape[1] == split_result["images"].shape[2]
+        ), "image width should be equal to image height"
+
+        cur_plocs, cur_source_type, cur_flux = get_lsst_params(
+            lsst_catalog_tensors_dict,
+            cur_image_wcs,
+            image_lim,
+            image_height_index=split_result["image_height_index"],
+            image_width_index=split_result["image_width_index"],
+        )
+
+        with open(output_path / ("lsst_" + split_id), "wb") as output_file:
+            torch.save(
+                {
+                    "plocs": cur_plocs.clone(),
+                    "source_type": cur_source_type.clone(),
+                    "flux": cur_flux.clone(),
```

Comment:
Please use an intermediate variable here, to avoid deep nesting

Reply:
fixed it